src/libs/ajax/WorkspaceDataService.test.ts

```diff
@@ -0,0 +1,123 @@
+import { asMockedFn } from '@terra-ui-packages/test-utils';
+import { Ajax } from 'src/libs/ajax';
+
+import { fetchWDS } from './ajax-common';
+import { Capabilities } from './WorkspaceDataService';
+
+type AjaxCommonExports = typeof import('src/libs/ajax/ajax-common');
+
+jest.mock('src/libs/ajax/ajax-common', (): AjaxCommonExports => {
+  return {
+    ...jest.requireActual('src/libs/ajax/ajax-common'),
+    fetchWDS: jest.fn(),
+  };
+});
+
+describe('WorkspaceData', () => {
+  describe('getCapabilities', () => {
+    type SetupOptions = {
+      stubbedCapabilitiesJson?: string;
+      stubbedCapabilitiesResponse?: Response;
+      wdsProxyUrl?: string;
+    };
+    type SetupResult = {
+      wdsProxyUrl: string;
+    };
+    function setup({
+      wdsProxyUrl = 'https://wds.test.proxy.url',
+      stubbedCapabilitiesJson = '{"capabilities":true}',
+      stubbedCapabilitiesResponse = new Response(stubbedCapabilitiesJson),
+    }: SetupOptions): SetupResult {
+      asMockedFn(fetchWDS).mockImplementation((wdsProxyUrlRoot: string) => {
+        if (wdsProxyUrlRoot !== wdsProxyUrl) {
+          throw new Error(`Unexpected wdsProxyUrlRoot: ${wdsProxyUrlRoot}`);
+        }
+        return (path: RequestInfo | URL, _options: RequestInit | undefined) => {
+          if (path === 'capabilities/v1') {
+            return Promise.resolve(stubbedCapabilitiesResponse);
+          }
+          throw new Error('Unexpected path');
+        };
+      });
+
+      return {
+        wdsProxyUrl,
+      };
+    }
+
+    it('returns true for capabilities when present and true', async () => {
+      // Arrange
+      const { wdsProxyUrl } = setup({ stubbedCapabilitiesJson: '{ "capabilities": true }' });
+
+      // Act
+      const capabilities = await Ajax().WorkspaceData.getCapabilities(wdsProxyUrl);
+
+      // Assert
+      expect(fetchWDS).toHaveBeenCalledWith(wdsProxyUrl);
+      expect(capabilities.capabilities).toEqual(true);
+    });
+
+    it('returns false for capabilities when present and false', async () => {
+      // Arrange
+      const { wdsProxyUrl } = setup({ stubbedCapabilitiesJson: '{ "capabilities": false }' });
+
+      // Act
+      const capabilities = await Ajax().WorkspaceData.getCapabilities(wdsProxyUrl);
+
+      // Assert
+      expect(fetchWDS).toHaveBeenCalledWith(wdsProxyUrl);
+      expect(capabilities.capabilities).toEqual(false);
+    });
+
+    it('returns false for capabilities when present and non-boolean', async () => {
+      // Arrange
+      const { wdsProxyUrl } = setup({ stubbedCapabilitiesJson: '{ "capabilities": "false" }' });
+
+      // Act
+      const capabilities: Capabilities = await Ajax().WorkspaceData.getCapabilities(wdsProxyUrl);
+
+      // Assert
+      expect(fetchWDS).toHaveBeenCalledWith(wdsProxyUrl);
+      expect(capabilities.capabilities).toEqual(false);
+    });
+
+    it('returns true for an unknown capability that is present and true', async () => {
```

Comment:
I don't know if this was actually desirable behavior, but I left it in as a form of documentation because I wasn't sure what to expect.

Reply:
@davidangb Yes exactly; I wasn't sure how TypeScript typing would work here and made this to explicitly demonstrate the behavior.